File: backend/src/main/java/com/rapidphoto/domain/PhotoStatus.java
package com.rapidphoto.domain;

/**
 * Status of photo processing lifecycle.
 */
public enum PhotoStatus {
    PENDING_PROCESSING,  // Photo uploaded, waiting for Lambda processing
    PROCESSING,          // Lambda is processing the photo
    READY,              // Photo fully processed and available
    FAILED              // Processing failed
}

File: mobile/lib/features/gallery/models/photo_status.dart
import 'package:json_annotation/json_annotation.dart';

/// Status of photo processing lifecycle
@JsonEnum(fieldRename: FieldRename.screamingSnake)
enum PhotoStatus {
  /// Photo uploaded, waiting for Lambda processing
  pendingProcessing,

  /// Lambda is processing the photo
  processing,

  /// Photo fully processed and available
  ready,

  /// Processing failed
  failed,
}
